test/e2e/index.ts

```diff
@@ -272,78 +273,86 @@ Leanplum.on('showMessage', (args) => {
-    title = message.Title.Text;
-    body = message.Message.Text;
+    title = message.Title.Text
+    body = message.Message.Text
 
     buttons.push(
       { text: message['Accept button'].Text, action: 'Accept action', primary: true }
     )
   } else {
     // unknown action, do not show
-    console.log(`Skipping unsupported (by Rondo) action: ${message.__name__}`);
-    return;
+    console.log(`Skipping unsupported (by Rondo) action: ${message.__name__}`)
+    return
   }
 
   const modalId = `lpModal-${message.messageId}${args.isPreview ? '-preview' : ''}`
 
-  const buttonHtml = button =>
-    `<button
-        class="btn btn-${ button.primary ? 'primary' : 'secondary' }"
-        data-action="${button.action}"
-        data-toggle="modal"
-        data-target="#${modalId}">
+  const getButtonHtml = (button): string => (`
+    <button
+      class="btn btn-${button.primary ? 'primary' : 'secondary'}"
+      data-action="${button.action}"
+      data-toggle="modal"
+      data-target="#${modalId}"
+    >
       ${button.text}
-    </button>`
-  const footer = !buttons.length ? '' :
-    `<div class="modal-footer">${buttons.map(buttonHtml).join('')}</div>`
+    </button>
+  `)
 
-  const header = !title ? '' :
-    `<div class="modal-header">
+  const footerHtml = buttons.length && (`
+    <div class="modal-footer">${buttons.map(getButtonHtml).join('')}</div>
+  `)
+
+  const headerHtml = title && (`
+    <div class="modal-header">
       <h5 class="modal-title">${title}</h5>
       <button type="button" class="close" data-dismiss="modal" aria-label="Close">
         <span aria-hidden="true">&times;</span>
       </button>
-    </div>`
-
-  const modal = `
-  <div class="modal" id="${modalId}" tabIndex="-1" role="dialog">
-    <form class="modal-dialog modal-dialog-centered" role="document">
-      <div class="modal-content">
-        ${header}
-        <div class="modal-body">
-          ${body}
+    </div>
+  `)
+
+  const modalHtml = (`
+    <div class="modal" id="${modalId}" tabIndex="-1" role="dialog">
+      <form class="modal-dialog modal-dialog-centered" role="document">
+        <div class="modal-content">
+          ${headerHtml || ''}
+          <div class="modal-body">
+            ${body}
+          </div>
+          ${footerHtml || ''}
         </div>
-        ${footer}
-      </div>
-    </form>
-  </div>
-`
+      </form>
+    </div>
+  `)
 
-  const runTrackedAction = (e) => {
+  const runTrackedAction = (e): void => {
     e.preventDefault()
     const action = $(e.currentTarget).data('action')
     if (!action) return
     context.runTrackedActionNamed(action)
   }
 
-  $(modal).hide().appendTo('body')
-    .on('shown.bs.modal', () => context.track())
-    .on('hidden.bs.modal', () => $(`#${modalId}`).remove())
-    .find('button').on('click', runTrackedAction).end()
-    .modal({ show: true })
+  const modal: (JQuery<HTMLDivElement> & { modal?: (options?) => JQuery<HTMLDivElement> }) =
```

Comment:
nit: these could probably be extracted as a single type, instead of used inline?

Reply:
Thanks! I moved them to type definitions file. This way they don't clutter the implementation.